arangod/Cluster/CreateCollection.h

```diff
@@ -33,7 +33,7 @@
 namespace arangodb {
 namespace maintenance {
 
-class CreateCollection : public ActionBase {
+class CreateCollection : public ActionBase, public ShardDefinition {
```

Comment:
As `~ShardDefinition()` isn't marked as `virtual`, the following code won't destroy `CreateCollection` object correctly 
```
ShardDefinition *p = new CreateCollection(...);
delete p;
```
I suggest to use aggregation instead of inheritance for this case.

Reply:
We don't have any code that will delete maintenance actions via a pointer to a ShardDefinition, and I don't think we will add it any soon.
Anyway, I can make it virtual easily. 
Using aggregation will also work but requires adjusting all call sites that access `getDatabase()` and `getShard()`. Possible, but more code, more work, and potentially not necessary.